### PR TITLE
[RV64_DYNAREC] Added more opcode and some fixes

### PR DESCRIPTION
**Pull request by ksco**
test01 ~ test16 runs fully in Dynarec now.

### CODE REVIEW

File: src/dynarec/rv64/dynarec_rv64_helper.c [hunk 1]
```diff
@@ -994,18 +994,19 @@ int x87_setround(dynarec_rv64_t* dyn, int ninst, int s1, int s2, int s3)
     // MMX/x87 Round mode: 0..3: Nearest, Down, Up, Chop
     // RV64: 0..7: Nearest, Toward Zero (Chop), Down, Up, Nearest tie to Max, invalid, invalid, dynamic (invalid here)
     // 0->0, 1->2, 2->3, 3->1
-    SLLI(s1, s1, 1);
```
Comment (ksco) [inline src/dynarec/rv64/dynarec_rv64_helper.c:997]: I can't figure out what the original version is trying to do, and it's problematic, so I replaced it with a more trivial version (which has three direct jumps).

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_helper.c:997]: Well, it's trying to change from 0, 1, 2, 3 to 0, 2, 3, 1 as the last comment says

File: src/dynarec/rv64/dynarec_rv64_df.c [hunk 2]
```diff
@@ -111,10 +111,26 @@ uintptr_t dynarec64_DF(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
                         FCVTDL(v1, x1, RD_RNE);
                     }
                     break;
+                case 1:
+                    INST_NAME("FISTTP Ew, ST0");
+                    v1 = x87_get_st(dyn, ninst, x1, x2, 0, EXT_CACHE_ST_F);
+                    addr = geted(dyn, addr, ninst, nextop, &wback, x3, x4, &fixedaddress, rex, NULL, 0, 0);
+                    FCVTWD(x4, v1, RD_RTZ);
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_df.c:118]: don't we need a check for Invalid Operation to change value to `0x8000` here?

File: src/dynarec/rv64/dynarec_rv64_df.c [hunk 3]
```diff
@@ -111,10 +111,26 @@ uintptr_t dynarec64_DF(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
...
+                    SH(x4, wback, fixedaddress);
+                    x87_do_pop(dyn, ninst, x3);
+                    break;
+                case 3: // TODO: fix rounding
+                    INST_NAME("FISTP Ew, ST0");
+                    v1 = x87_get_st(dyn, ninst, x1, x2, 0, EXT_CACHE_ST_F);
+                    addr = geted(dyn, addr, ninst, nextop, &wback, x3, x4, &fixedaddress, rex, NULL, 0, 0);
+                    FCVTWD(x4, v1, RD_RTZ);
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_df.c:126]: Same

File: src/dynarec/rv64/dynarec_rv64_df.c (hunk 2 above)
Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_df.c:118]: Oh, the interpreter seems not doing it on this one, I'll add that on both sides.

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_df.c:118]: Ah, it's probably missing there too them. I would expect all conversion to int to react the same way on invalid operations

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_df.c:118]: Emmm, can I just use `FR_NV` on this? It will need a lot of opcodes if not.

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_df.c:118]: Start with that, and encased the code in a `if(!box64_dynarec_fastround)` yes.

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_df.c:121]: I've pushed a new commit, but since this is a `double -> int16` conversion, but we only have `FCVTWD` at hand, it's not accurate, right?

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_df.c:121]: Yep, there should be another test when doing 32bits -> 16bits reduction, to check for overflow there too

File: src/dynarec/rv64/dynarec_rv64_df.c [hunk 4]
```diff
@@ -123,6 +123,8 @@ uintptr_t dynarec64_DF(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
                         FRFLAGS(x5);   // get back FPSR to check the IOC bit
                         ANDI(x5, x5, 1<<FR_NV);
                         CBZ_NEXT(x5);
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_df.c:125]: CBZ_NEXT? So the next test will never be done

File: src/dynarec/rv64/dynarec_rv64_df.c [hunk 5]
```diff
@@ -123,6 +123,8 @@ uintptr_t dynarec64_DF(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
...
+                        SRLIW(x5, x4, 16);
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_df.c:126]: You are sure this will work for negative values. I think it will never be 0 because the high part will be filled with 0xffffffff after the `FCVTWD`?

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_df.c]: Sorry, I've force-pushed, I believe it's right this time.
